watertap/property_models/NaCl_T_dep_prop_pack.py

```diff
@@ -0,0 +1,1797 @@
+        )
+
+        def rule_flow_mol_phase_comp(b, p, j):
+            return (
+                b.flow_mol_phase_comp[p, j]
+                == b.flow_mass_phase_comp[p, j] / b.params.mw_comp[j]
+            )
+
+        self.eq_flow_mol_phase_comp = Constraint(
+            self.params.phase_list,
+            self.params.component_list,
+            rule=rule_flow_mol_phase_comp,
+        )
+
+    def _mole_frac_phase_comp(self):
+        self.mole_frac_phase_comp = Var(
+            self.params.phase_list,
+            self.params.component_list,
+            initialize=0.1,
+            bounds=(0.0, None),
+            units=pyunits.dimensionless,
+            doc="Mole fraction",
+        )
+
+        def rule_mole_frac_phase_comp(b, p, j):
+            return b.mole_frac_phase_comp[p, j] == b.flow_mol_phase_comp[p, j] / sum(
+                b.flow_mol_phase_comp[p, j] for j in b.params.component_list
+            )
+
+        self.eq_mole_frac_phase_comp = Constraint(
+            self.params.phase_list,
+            self.params.component_list,
+            rule=rule_mole_frac_phase_comp,
+        )
+
+    def _molality_phase_comp(self):
+        self.molality_phase_comp = Var(
+            self.params.phase_list,
+            ["NaCl"],
+            initialize=1,
+            bounds=(1e-4, 10),
+            units=pyunits.mole / pyunits.kg,
+            doc="Molality",
+        )
+
+        def rule_molality_phase_comp(b, p, j):
+            return (
+                self.molality_phase_comp[p, j]
+                == b.mass_frac_phase_comp[p, j]
+                / (1 - b.mass_frac_phase_comp[p, j])
+                / b.params.mw_comp[j]
+            )
+
+        self.eq_molality_phase_comp = Constraint(
+            self.params.phase_list, ["NaCl"], rule=rule_molality_phase_comp
+        )
+
+    def _solubility_comp(self):
+        self.solubility_comp = Var(
+            ["NaCl"],
+            initialize=0.5,
+            bounds=(0.0, 1.01),
+            units=pyunits.dimensionless,
+            doc="solubility_comp",
+        )
+
+        def rule_solubility_comp(b, j):
+            t = (b.temperature - 273.15 * pyunits.K) / pyunits.K
+            return (
+                self.solubility_comp[j]
+                == b.params.solubility_comp_param["0"]
+                + b.params.solubility_comp_param["1"] * t
+                + b.params.solubility_comp_param["2"] * t**2
+            )
+
+        self.eq_solubility_comp = Constraint(["NaCl"], rule=rule_solubility_comp)
+
+    def _pressure_sat(self):
+        self.pressure_sat = Var(
+            initialize=0.01,
+            bounds=(0.0, 5),
+            units=pyunits.MPa,
+            doc="Vapor Pressure",
+        )
+
+        def rule_pressure_sat(b):  # vapor pressure, eq. 6 in Sparrow 2003
+            t = (b.temperature - 273.15 * pyunits.K) / pyunits.K
+            T = value(t)
+            x = b.mass_frac_phase_comp["Liq", "NaCl"]
+            if (T >= 0) and (T <= 150):
```

Comment:
I would be careful with this since temperature is a variable, and during a solve, the value would be changing with each iteration. One could argue that temperature can typically be fixed, but since temp is indeed a variable and can't be guaranteed to always be fixed, I don't think you should use if-else conditionals.

Instead, you can try an `Expr_if`, which essentially should do what you intended the if-else conditional to do for you but as an `Expression`. The caveat is that instability can supposedly occur when the variable approaches the bound (e.g., T =150), but I haven't encountered this issue myself.

Reply:
You should generally never do this in a rule - the rule is only evaluated once at the time of model construction, and so will be based on the initial value of T only (which is likely a default value at this point). Also note that it is generally a very bad idea to use `value` inside rule.

As @adam-a-a said, you can use an `Expr_If` if a conditional is required or a smooth approximation of the if. However, you should always ask yourself if the `if` even matters - i.e. in the actual application will the `if` ever trigger?

Reply:
I played around with this today. The `Expr_if` did work for what I had wanted it to do, but as you said (@adam-a-a), it did have difficulty with T = 150.

However, after giving your question (@andrewlee94) some thought, I don't think the `if` matters. I am unaware of applications which go above 150 C. Please correct me if I am forgetting an example which does.

Reply:
I updated the pressure_sat with new parameters from a regression using new data which will avoid the need for an if statement while also encompassing the full temperature range up to 300 C.